R/utilities-scenarios.R

```diff
@@ -2,311 +2,149 @@
-        error = function(cond) {
-          warning(paste0("Cannot save to path '", outputFolder, "'"))
-          message("Original error message:")
-          message(cond)
-        },
-        warning = function(cond) {
-          warning(cond)
-        }
-      )
+    if (scenario$scenarioConfiguration$simulateSteadyState) {
+      steadyStateSimulations <- c(steadyStateSimulations, scenario$simulation)
+      steadyStateTimes <- c(steadyStateTimes, scenario$scenarioConfiguration$steadyStateTime)
     }
+  }
 
-    # Save population to CSV
-    if (savePopulationToCSV) {
-      outputFolder <- file.path(
-        scenarioConfiguration$projectConfiguration$outputFolder,
-        "Populations",
-        outputFolderSuffix
-      )
-      # Create a new folder if it does not exist
-      if (!dir.exists(paths = outputFolder)) {
-        dir.create(path = outputFolder, recursive = TRUE)
-      }
-      if (!is.null(population)) {
-        ospsuite::exportPopulationToCSV(population$population,
-          filePath = file.path(outputFolder, paste0(scenarioName, "population.csv"))
-        )
-      }
-    }
+  # Simulate steady-state concurrently
+  if (length(steadyStateSimulations) > 0) {
+    initialValues <- ospsuite.parameteridentification::getSteadyState(
+      simulations = steadyStateSimulations,
+      steadyStateTime = steadyStateTimes, ignoreIfFormula = TRUE,
+      simulationRunOptions = simulationRunOptions
+    )
   }
-  names(simulations) <- scenarioNames
-  names(scenarioConfigurations) <- scenarioNames
-  names(populations) <- scenarioNames
 
-  # Simulate individual simulations concurrently
-  individualSimulationsIdx <- is.na(populations)
+  # Set initial values for steady-state simulations
+  for (simulation in steadyStateSimulations) {
+    ospsuite::setQuantityValuesByPath(
+      quantityPaths = initialValues[[simulation$id]]$paths,
+      values = initialValues[[simulation$id]]$values, simulation = simulation
+    )
+  }
+
+  # Run invidual simulations
   simulationResults <- runSimulations(
-    simulations = simulations[individualSimulationsIdx],
-    simulationRunOptions = scenarioConfiguration$simulationRunOptions
+    simulations = individualSimulations,
+    simulationRunOptions = simulationRunOptions
   )
 
   # Run population simulations sequentially and add the to the list of simulation results
-  for (scenarioName in scenarioNames[!individualSimulationsIdx]) {
+  for (scenario in populationScenarios) {
     populationResults <- runSimulations(
-      simulations = simulations[[scenarioName]],
-      population = populations[[scenarioName]],
-      simulationRunOptions = scenarioConfiguration$simulationRunOptions
+      simulations = scenario$simulation,
+      population = scenario$population,
+      simulationRunOptions = simulationRunOptions
     )
     simulationResults <- c(simulationResults, populationResults)
   }
 
+  # Create output list with simulation results, simulation objects, and population objects
```

Comment:
Maybe this step can be done together with the other steps (there's already a for loop on `scenarios` earlier in code).

Reply:
I don't think so, we don't have `simulationResults` up to this step.